src/figuresClasses.ts

```diff
@@ -1,19 +1,97 @@
-export interface Figure {
+type ShapeOfFigure = 'triangle' | 'circle' | 'rectangle';
+type ColorOfFigure = 'red' | 'green' | 'blue';
 
+function roundDownToHundredths(value: number): number {
+  return Math.floor(value * 100) / 100;
 }
 
-export class Triangle {
+function sideSmallerZeroError(side: number, shape: ShapeOfFigure): void {
+  if (side <= 0) {
+    switch (shape) {
+      case 'circle':
+        throw new Error(`It’s even hard for me to imagine
+        what kind of triangle this should be.
+        I'm not a quantum computer, can I have a normal triangle?`);
+
+      case 'rectangle':
+        throw new Error(`It’s even hard for me to imagine
+        what kind of triangle this should be.
+        I'm not a quantum computer, can I have a normal triangle?`);
 
+      default:
+    }
+  }
 }
 
-export class Circle {
+function longestSideOfTriangleBiggerError(sides: number[]): void {
+  if (sides.sort((prev, cur) => prev - cur)[2]
+    >= sides.sort((prev, cur) => prev - cur)[0]
+    + sides.sort((prev, cur) => prev - cur)[1]) {
```

Comment:
Hard to read, create variables. Also, do you need to sort three times? Can you sort just one time?